server/src/ClientToDBAPI/UsersRoute/mainRoute.ts

```diff
@@ -219,32 +220,32 @@ usersRoute.get('/group', adminMiddleWare, (request: Request, response: Response)
 });
 
 usersRoute.post('/changeCounty', adminMiddleWare, (request: Request, response: Response) => {
+    const epidemiologyNumbers: number[] = request.body.epidemiologyNumbers;
     let userAdmin = 'admin.group' + request.body.updatedCounty;
-    const workflow = 'GraphQL request to the change the investigation county';
+    const transferReason: string = request.body.transferReason || '';
     const changeCountyLogger = logger.setup({
-        workflow,
+        workflow: 'GraphQL request to the change the investigation county',
         user: response.locals.user.id,
-        investigation: response.locals.epidemiologyNumber
+        investigation: response.locals.epidemiologynumber,
     });
-    changeCountyLogger.info(`post with parameters ${JSON.stringify({ epidemiologyNumber: request.body.epidemiologyNumber, newUser: userAdmin })}`, Severity.LOW);
-
-    graphqlRequest(UPDATE_COUNTY_BY_USER, response.locals, {
-        epidemiologyNumber: request.body.epidemiologyNumber,
-        newUser: userAdmin
-    }).then((result: any) => {
-        if (result?.data && !result?.errors) {
-            changeCountyLogger.info('The investigation county changed successfully', Severity.LOW);
-            response.send({message: 'The county has changed successfully'});
+    changeCountyLogger.info('launching graphql API counties request', Severity.LOW);
+    Promise.all(epidemiologyNumbers.map(epidemiologyNumber => graphqlRequest(UPDATE_COUNTY_BY_USER, response.locals, {
```

Comment:
why mapping the epidemiologyNumbers and execute UPDATE_COUNTY_BY_USER for each epidemiologyNumber
why not changing the query (trying to use IN syntax in sql)